packages/pricing-calculator/src/calculator/calculator.component.tsx

```diff
@@ -28,11 +28,26 @@ import {
 } from './calculator.config';
 import { servicesConfig } from '../services.config';
 
-import { updateService, updateQueries, updateEvents } from './actions';
+import {
+  updateService,
+  updateQueries,
+  updateEvents,
+  setSliderDimension,
+} from './actions';
+import { getCalculatorState } from './selectors';
+
+import { convertPositionToValue } from '../utils';
 
 const Calculator: FC<{}> = () => {
   const dispatch = useDispatch();
   const device = useSelector(getDevice);
+  const { sliderDimension, events, queries } = useSelector(getCalculatorState);
+  const sliderRef = useRef(null);
+  useEffect(() => {
+    if (sliderRef.current) {
+      dispatch(setSliderDimension(sliderRef.current.offsetWidth));
```

Comment:
Why the dimension of `Slider` is part of Redux store ? Do we need to share this information between all application components ?